inc/c_util/two_d_array_ll.h

```diff
@@ -15,10 +15,236 @@
+            }                                                                                                                                                                                \
+            else                                                                                                                                                                             \
+            {                                                                                                                                                                                \
+                /* Codes_SRS_TWO_D_ARRAY_07_016: [ If the row specified by row_index has already been allocated, TWO_D_ARRAY_ALLOCATE_NEW_ROW(T) shall fail and return a non-zero value. ]*/ \
+                result = MU_FAILURE;                                                                                                                                                         \
+            }                                                                                                                                                                                \
+        }                                                                                                                                                                                    \
+    }                                                                                                                                                                                        \
+    return result;                                                                                                                                                                           \
+}
+
+#define TWO_D_ARRAY_LL_GET_DEFINE(C, T)                                                                                                                    \
+T* TWO_D_ARRAY_LL_GET(C)(TWO_D_ARRAY_LL(T) two_d_array, uint64_t row_index)                                                                                \
+{                                                                                                                                                          \
+    T* result;                                                                                                                                             \
+    /* Codes_SRS_TWO_D_ARRAY_07_019: [ If two_d_array is NULL, TWO_D_ARRAY_GET(T) shall fail return NULL. ]*/                                              \
+    if (two_d_array == NULL)                                                                                                                               \
+    {                                                                                                                                                      \
+        LogError("Invalid arguments: TWO_D_ARRAY_HANDLE two_d_array=%p", two_d_array);                                                                     \
```

Comment:
>TWO_D_ARRAY_HANDLE  [](http://example.com/codeflow?start=37&length=19)

All the loglines should printf 
"TWO_D_ARRAY(" MU_TOSTRING(T) ")"

so that we know what we're dealing with as structures rather than a generic TWO_D_ARRAY_HANDLE which does not even exist :-)

Reply:
changed to use MU_TOSTRING here and everywhere else